testutil/ciao-down/cloudinit.go

```diff
@@ -94,7 +94,7 @@ runcmd:
  - echo "PATH=\"$PATH:/usr/local/go/bin:{{$.GoPath}}/bin:/usr/local/nodejs/bin\""  >> /etc/environment
 
  - curl -X PUT -d "Downloading Go" 10.0.2.2:{{.HTTPServerPort}}
- - {{template "ENV" .}}wget https://storage.googleapis.com/golang/go1.8.linux-amd64.tar.gz -O /tmp/go1.8.linux-amd64.tar.gz
```

Comment:
> If this were split into an url and file in a const struct like with guestDownloadURL, there wouldn't be duplication of the strings for the filename (eg: "go1.8.linux-amd64.tar.gz").

I could add a downloadTo function which could be used when the last part of the url matches the desired filename.  That way you'd only need to specify a directory, e.g.,

```
{{downloadTo . "https://storage.googleapis.com/golang/go1.8.linux-amd64.tar.gz" "/tmp"}}
```


> It would be better to not do this in something world writeable like /tmp, versus ~/local.

Okay

> Is there a reason to use both curl and wget here? 

To be honest I don't remember but I agree it would make sense to standardise on one.  Going forward these programs should mostly be hidden from the script writer.  Curl is currently used to send status information back to ciao-down.  This task will soon be replaced by a function.

```
- curl -X PUT -d "Downloading Go" 10.0.2.2:{{.HTTPServerPort}}
```

will become

```
- {{info . "Downloading Go"}}
```